File: lab2/c/main.c
#include <stdio.h>
#include <time.h>
#include <stdlib.h>
#include "bubble_sort.h"
#include "insertion_sort.h"
#include "merge_sort.h"

void print_array(int *A, int N) {
    printf("A: [");
    for (int i = 0; i < N; ++i) printf(" %d", A[i]);
    printf(" ]\n");
}

void test_bubble_sort() {
    printf("\nBubble Sort:\n");
    int N = rand() % 15 + 1; // Generate a random number between [1, 15]
    int A[N];
    for (int i = 0; i < N; ++i) A[i] = rand() % 101; // Assign a random number between [0, 100]

    printf("Before sort: ");
    print_array(A, N);

    bubble_sort(A, N);
    printf("After sort: ");
    print_array(A, N);
}

void test_insertion_sort() {
    printf("\nInsertion Sort:\n");
    int N = rand() % 15 + 1; // Generate a random number between [1, 15]
    int A[N];
    for (int i = 0; i < N; ++i) A[i] = rand() % 101; // Assign a random number between [0, 100]

    printf("Before sort: ");
    print_array(A, N);

    insertion_sort(A, N);
    printf("After sort: ");
    print_array(A, N);
}

void test_merge_sort() {
    printf("\nMerge Sort:\n");
    int N = rand() % 15 + 1; // Generate a random number between [1, 15]
    int A[N];
    for (int i = 0; i < N; ++i) A[i] = rand() % 101; // Assign a random number between [0, 100]

    printf("Before sort: ");
    print_array(A, N);

    merge_sort(A, 0, N-1);
    printf("After sort: ");
    print_array(A, N);
}

int main() {
    srand(time(NULL));

    test_bubble_sort();
    test_insertion_sort();
    test_merge_sort();
}

File: lab2/python/main.py
import random
from bubble_sort import bubble_sort
from insertion_sort import insertion_sort
from merge_sort import merge_sort

def test_bubble_sort():
    print('\nBubble Sort: ')
    N = random.randint(1, 15);
    A = []      # This is similar to declaring empty array (called list in python)
    for i in range(0, N):
        A.append(random.randint(0, 100))

    print(f'Before sort: {A}')
    bubble_sort(A)
    print(f'After sort: {A}')

def test_insertion_sort():
    print('\nInsertion Sort: ')
    N = random.randint(1, 15);
    A = []      # This is similar to declaring empty array (called list in python)
    for i in range(0, N):
        A.append(random.randint(0, 100))

    print(f'Before sort: {A}')
    insertion_sort(A)
    print(f'After sort: {A}')

def test_merge_sort():
    print('\nMerge Sort: ')
    N = random.randint(1, 15);
    A = []      # This is similar to declaring empty array (called list in python)
    for i in range(0, N):
        A.append(random.randint(0, 100))

    print(f'Before sort: {A}')
    merge_sort(A, 0, N-1)
    print(f'After sort: {A}')



if __name__ == '__main__':
    test_bubble_sort()
    test_insertion_sort()
    test_merge_sort()
